### PR TITLE
Fixed JS that was loading message in wrong order. (Issue #728)

### PR DESCRIPTION
**Pull request by antonini**
Fixed JS that was loading message in wrong order. (Issue #728).

### CODE REVIEW

**Comment by karato** [comment]
@antonini I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by karato** [comment]
@pinaf do this review

**Comment by pinaf** [comment]
@antonini travis is not building

**Comment by antonini** [comment]
@pinaf Fixed.

File: netbout-web/src/test/casperjs/navigate-to-old-message-on-bout.js [hunk 1]
```diff
@@ -119,6 +119,48 @@ casper.test.begin(
...
+                casper.log(response.url,"error");
+                var dates = this.getElementsAttribute(
+                    "#messages .message a span",
+                    "title"
+                );
+                var lastDate = null;
+                for (var i = 0; i < dates.length; i++) {
+                    date = Date.parse(dates[i]);
+                    if (!lastDate) {
+                        this.echo(
```
Comment (pinaf) [inline netbout-web/src/test/casperjs/navigate-to-old-message-on-bout.js]: @antonini are these calls to 'echo' just for testing?

**Comment by pinaf** [comment]
@antonini 1 question

File: netbout-web/src/test/casperjs/navigate-to-old-message-on-bout.js (hunk 1 above)
Comment (antonini) [inline netbout-web/src/test/casperjs/navigate-to-old-message-on-bout.js]: @pinaf Ops... Fixed it.

**Comment by antonini** [comment]
@pinaf Ready.

File: netbout-web/src/main/js/bout.js [hunk 2]
```diff
@@ -135,18 +135,21 @@ function readMore(retFunction) {
...
-    // @todo #643:30min/DEV System should highlights a message when accessing it
-    //  through message's permalink. This must be implemented on a different
-    //  method and must be called over here and on the `else` statement of the
-    //  same `if else` block.
-    $(window).scrollTop($(location.hash.trim()).offset().top);
-  } else if (location.hash.trim() && amountRecords > 0) {
-    readMore(scrollOrLoad);
-  } else {
-    $(window).scrollTop($(location.hash.trim()).offset().top);
+  if (location.hash.trim()) {
```
Comment (pinaf) [inline netbout-web/src/main/js/bout.js]: @antonini This is just to make it more readable, right? No functionality was changed?

File: netbout-web/src/test/casperjs/navigate-to-old-message-on-bout.js [hunk 3]
```diff
@@ -119,6 +119,41 @@ casper.test.begin(
...
+                      headers: {
+                        'Accept': 'text/html'
+                      }
+                    }
+                );
+            }
+        );
+        casper.then(
+            function(response) {
+                casper.log(response.url,"error");
```
Comment (pinaf) [inline netbout-web/src/test/casperjs/navigate-to-old-message-on-bout.js]: @antonini what about this one?

File: netbout-web/src/test/casperjs/navigate-to-old-message-on-bout.js [hunk 4]
```diff
@@ -119,6 +119,41 @@ casper.test.begin(
...
+                );
+                var lastDate = null;
+                for (var i = 0; i < dates.length; i++) {
+                    date = Date.parse(dates[i]);
+                    if (!lastDate) {
+                        lastDate = date;
+                    } else {
+                        test.assert(
+                            (lastDate >= date),
+                            "It was loaded an older message before a new one."
```
Comment (pinaf) [inline netbout-web/src/test/casperjs/navigate-to-old-message-on-bout.js]: @antonini `An older message was loaded before a newer message.`

File: netbout-web/src/main/js/bout.js [hunk 5]
```diff
@@ -135,18 +135,21 @@ function readMore(retFunction) {
...
+        ($(msgId).offset().top <
+            $(document).height() - $(window).height() - 600)) {
+      // @todo #643:30min/DEV System should highlights a message when accessing it
+      //  through message's permalink. This must be implemented on a different
+      //  method and must be called over here and on the `else` statement of the
+      //  same `if else` block.
+      $(window).scrollTop($(msgId).offset().top);
+    } else if (amountRecords > 0) {
+      readMore(scrollOrLoad);
+    } else if ($(msgId).length >= 1) {
```
Comment (pinaf) [inline netbout-web/src/main/js/bout.js]: @antonini I guess this is the behavior fix, correct?

**Comment by pinaf** [comment]
@antonini just 1 more comment on the error message text.

**Comment by antonini** [comment]
@pinaf Changed message. Ready for review.

**Comment by pinaf** [comment]
@rultor merge

**Comment by pinaf** [comment]
@antonini thanks

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by dmzaytsev** [comment]
@rultor deploy

**Comment by karato** [comment]
@elenavolokhova please, review this ticket for compliance with our [QA rules](http://www.teamed.io/qa.html)

**Comment by karato** [comment]
@rultor deploy

**Comment by elenavolokhova** [comment]
@pinaf 
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 There were only two issues found during review.
Please confirm that you will try to find at least three problems next time.

**Comment by pinaf** [comment]
@elenavolokhova I confirm

**Comment by elenavolokhova** [comment]
@karato Quality is acceptable.

**Comment by karato** [comment]
> @karato Quality is acceptable.

@elenavolokhova thanks for the QA review, we'll work better next time

**Comment by karato** [comment]
@pinaf I added 10 mins to @elenavolokhova (for QA review) in transaction 60886384. Much appreciated! **21 mins** was added to your account, payment ID `AP-9YF83997P1098892K`, time spent is 45 hours and 50 mins. you're getting extra minutes here (c=6). added +21 to your rating, now it is equal to [+6568](http://www.netbout.com/b/35128?open=rating)

**Pull request merged**